7_react/react1/week3/todo-app/src/components/TodoForm/TodoForm.jsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import { generateId } from "./utilities";
+import DateSelector from "../DatePicker/DatePicker";
+import './TodoForm.css';
+
+
+export function TodoForm(props) {
+    const [description, setDescription] = useState("");
+    const [selectedDate, setSelectedDate] = useState(null);
+
+    const handleTextChange = ({ target }) => {
+        const { value } = target;
+        setDescription(value);
+    };
+
+    const handleDateChange = (date) => {
+        setSelectedDate(date);
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (description.trim() && selectedDate) {
+            const todo = {
+                id:generateId(),
+                description: description,
+                deadline: selectedDate.toISOString().split('T')[0],
```

Comment:
This will also cause some issues with in your timezone as well. If you've noticed, the displayed date will differ from the selected one by one day. This is caused by the `toISOString` method, which converts the date to UTC time. Alternatively, you can use this approach if you know that only people from Denmark will use it.

```
selectedDate.toLocaleDateString("en-CA", {
        timeZone: "Europe/Copenhagen",
      });

```